# hypothesis testing

## Introduction

This environmental think tank is called Repair Our Air (ROA). ROA is formulating policy recommendations to improve the air quality in America, using the Environmental Protection Agency's Air Quality Index (AQI) to guide their decision making. An AQI value close to 0 signals "little to no" public health concern, while higher values are associated with increased risk to public health.

The assignment is  with leveraging AQI data to help them prioritize their strategy for improving air quality in America.

ROA is considering the following decisions. For each, construct a hypothesis test and an accompanying visualization, using  results of that test to make a recommendation:

1. ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.
2. With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?
3. A new policy will affect those states with a mean AQI of 10 or greater. Would Michigan be affected by this new policy?

**Notes:**
1. For  analysis,  default significance is 5% level of significance.
2. Throughout the lab, for two-sample t-tests, use Welch's t-test (i.e., setting the `equal_var` parameter to `False` in `scipy.stats.ttest_ind()`). This will account for the possibly unequal variances between the two groups in the comparison.

## Imports

#### Import Packages

In [1]:
# Import relevant packages
import pandas as pd
import numpy as np
from scipy import stats

Assumption: This dataset is randomly sampled from a larger population.

#### Loading Dataset

In [3]:
# Use read_csv() to import
aqi = pd.read_csv('/content/sample_data/c4_epa_air_quality.csv')

## Data Exploration

In [4]:
# Explore dataframe `aqi` here:
print("Use head() to show a sample of data")
print(aqi.head())

print("Use describe() to summarize AQI")
print(aqi.describe(include='all'))

print("For a more thorough examination of observations by state use values_counts()")
print(aqi['state_name'].value_counts())

print('for a more')

Use head() to show a sample of data
   Unnamed: 0  date_local    state_name   county_name      city_name  \
0           0  2018-01-01       Arizona      Maricopa        Buckeye   
1           1  2018-01-01          Ohio       Belmont      Shadyside   
2           2  2018-01-01       Wyoming         Teton  Not in a city   
3           3  2018-01-01  Pennsylvania  Philadelphia   Philadelphia   
4           4  2018-01-01          Iowa          Polk     Des Moines   

                                     local_site_name   parameter_name  \
0                                            BUCKEYE  Carbon monoxide   
1                                          Shadyside  Carbon monoxide   
2  Yellowstone National Park - Old Faithful Snow ...  Carbon monoxide   
3                             North East Waste (NEW)  Carbon monoxide   
4                                          CARPENTER  Carbon monoxide   

    units_of_measure  arithmetic_mean  aqi  
0  Parts per million         0.473684    7  
1 

#### **From preceding data exploration,the facts to be recognized**

- county-level data is available for the first hypothesis.
- Ohio and New York both have a higher number of observations to work with in this dataset.

## Statistical Tests

Following steps for conducting hypothesis testing:

1. Formulate the null hypothesis and the alternative hypothesis.<br>
2. Set the significance level.<br>
3. Determine the appropriate test procedure.<br>
4. Compute the p-value.<br>
5. Draw your conclusion.

### Hypothesis 1: ROA is considering a metropolitan-focused approach. Within California, they want to know if the mean AQI in Los Angeles County is statistically different from the rest of California.

In [5]:
# Create dataframes for each sample being compared in your test
ca_la = aqi[aqi['county_name']=='Los Angeles']
ca_other = aqi[(aqi['state_name']=='California') & (aqi['county_name']!='Los Angeles')]

Formulation of null and alternative hypotheses:

*   $H_0$: There is no difference in the mean AQI between Los Angeles County and the rest of California.
*   $H_A$: There is a difference in the mean AQI between Los Angeles County and the rest of California.


#### Setting significance level:

In [6]:
# For this analysis, the significance level is 5%

significance_level = 0.05
significance_level

0.05

#### Determination of the appropriate test procedure:

Comparing the sample means between two independent samples will utilize **two-sample  𝑡-test**.

#### Computing the p-value

In [7]:
# Compute your p-value here
stats.ttest_ind(a=ca_la['aqi'], b=ca_other['aqi'], equal_var=False)

TtestResult(statistic=2.1107010796372014, pvalue=0.049839056842410995, df=17.08246830361151)

#### **Relevant Question . What is p-value for hypothesis 1, and what does this indicate for null hypothesis?**

With a p-value (0.049) being less than 0.05 (as your significance level is 5%), reject the null hypothesis in favor of the alternative hypothesis.

Therefore, a metropolitan strategy may make sense in this case.

### Hypothesis 2: With limited resources, ROA has to choose between New York and Ohio for their next regional office. Does New York have a lower AQI than Ohio?

In [8]:
# Create dataframes for each sample being compared in your test
ny = aqi[aqi['state_name']=='New York']
ohio = aqi[aqi['state_name']=='Ohio']

#### Formulation of hypothesis:

**Formulation of my null and alternative hypotheses:**

*   $H_0$: The mean AQI of New York is greater than or equal to that of Ohio.
*   $H_A$: The mean AQI of New York is **below** that of Ohio.


#### Significance Level (remains at 5%)

#### Computing the p-value

In [9]:
# Compute your p-value here
tstat, pvalue = stats.ttest_ind(a=ny['aqi'], b=ohio['aqi'], alternative='less', equal_var=False)
print(tstat)
print(pvalue)

-2.025951038880333
0.03044650269193468


#### **Relavent Question. What is p-value for hypothesis 2, and what does this indicate for null hypothesis?**

With a p-value (0.030) of less than 0.05 (as significance level is 5%) and a t-statistic < 0 (-2.036), **reject the null hypothesis in favor of the alternative hypothesis**.

Therefore, it can be concluded at the 5% significance level that New York has a lower mean AQI than Ohio.

###  Hypothesis 3: A new policy will affect those states with a mean AQI of 10 or greater. Will Michigan be affected by this new policy?

In [10]:
# Create dataframes for each sample being compared in your test
michigan = aqi[aqi['state_name']=='Michigan']

#### Formulatio of  hypothesis:

**Formulate  null and alternative hypotheses here:**

*   $H_0$: The mean AQI of Michigan is less than or equal to 10.
*   $H_A$: The mean AQI of Michigan is greater than 10.


#### Significance Level (remains at 5%)

#### Compute the P-value

In [11]:
# Compute p-value here
tstat, pvalue = stats.ttest_1samp(michigan['aqi'], 10, alternative='greater')
print(tstat)
print(pvalue)

-1.7395913343286131
0.9399405193140109


#### **Question 4. What is p-value for hypothesis 3, and what does this indicate for  null hypothesis?**

With a p-value (0.940) being greater than 0.05 (as significance level is 5%) and a t-statistic < 0 (-1.74), **fail to reject the null hypothesis**.

Therefore, it cannot be concluded at the 5% significance level that Michigan's mean AQI is greater than 10. This implies that Michigan would most likely not be affected by the new policy.

## Step 4. Results and Evaluation

The results indicated that the AQI in Los Angeles County was in fact different from the rest of California.

Using a 5% significance level, it can be concluded that New York has a lower AQI than Ohio based on the results.

#### **Relavent Question: Will Michigan be affected by the new policy impacting states with a mean AQI of 10 or greater?**

Based on the tests, the conclusion can be failing to reject the null hypothesis, meaning it can't be concluded that the mean AQI is greater than 10. Thus, it is unlikely that Michigan would be affected by the new policy.



# Conclusion

**key takeaways from this lab**

Even with small sample sizes, the variation within the data is enough to allow you to make statistically significant conclusions. It is identified at the 5% significance level that the Los Angeles mean AQI was stastitically different from the rest of California, and that New York does have a lower mean AQI than Ohio. However, it cannot be concluded at the 5% significance level that Michigan's mean AQI was greater than 10.